Added support for transformed response to aov_car

rvlenth opened the pull request:
Henrik,

I added some code to `aov_car` that supports a response transformation.  If there is a transformation, the dv is computed and renamed, and the transformation is saved as the `"transf"` attribute of the returned object. The `emm_basis` method checks for this attribute and returns it appropriately if present. Thus, the response transformation is seamlessly integrated into subsequent calls to `emmeans`, be it univariate or multivariate.

NOTE: The anova output, etc. is done for the transformed variable. The variable name used (and displayed in the summary) is composed of the transformation and the response name; e.g., `sqrt(count) ~ ...` creates a transformed response named `sqrt.count`.

I did not add this support to `aov_ez` as I'm not sure how you'd want it implemented. The most natural thing seems to be to add, say, a `transform` argument that if non-missing, would incorporate it in the formula that is passed to `aov_car`. For example, `aov_ez(dv = "yield", transform = "sqrt")` creates the formula `sqrt(yield) ~ ...`. But another option might be just to support, e.g., `dv = "sqrt(yield)"`.

Example (based on the help page):
```
data(obk.long, package = "afex")
oblog <- aov_car(log(value) ~ treatment * gender + Error(id/(phase*hour)), 
                 data = obk.long, observed = "gender")
## Contrasts set to contr.sum for the following variables: treatment, gender

ref_grid(oblog)
## 'emmGrid' object with variables:
##     treatment = control, A, B
##     gender = F, M
##     phase = fup, post, pre
##     hour = X1, X2, X3, X4, X5
## Transformation: “log” 

ref_grid(oblog, model = "mult")
## 'emmGrid' object with variables:
##     treatment = control, A, B
##     gender = F, M
##     hour = multivariate response levels: X1, X2, X3, X4, X5
##     phase = multivariate response levels: fup, post, pre
## Transformation: “log” 
```

singmann commented:
Hi Russ,

Thanks a lot, that looks awesome. It would be great if you could also add the support for `aov_ez` via the additional `transform` argument. As you suggest this seems indeed the more natural way for this interface. If you can do this, I will wait with accepting the pull request until this is added.

Thanks a lot!

rvlenth commented:
OK, I added that argument. I also added documentation for it. You may want to place the argument earlier or later, but where I put it seems appropriate.

Additional example:
```
aov_ez("id", "value", obk.long, between = c("treatment", "gender"), 
       within = c("phase", "hour"), observed = "gender", transf = "sqrt", 
       print.formula = TRUE) -> obsqrt

## Formula send to aov_car: sqrt(value) ~ treatment * gender + Error(id/(phase * hour))
## Contrasts set to contr.sum for the following variables: treatment, gender

ref_grid(obsqrt)

## 'emmGrid' object with variables:
##     treatment = control, A, B
##     gender = F, M
##     phase = fup, post, pre
##     hour = X1, X2, X3, X4, X5
## Transformation: “sqrt” 
```

rvlenth commented:
I see that the Travis check failed. This happened because `devtools::document()` did not fully update the documentation in `aov_car.Rd`. It added the documentation from `@param transformation` but failed to update the argument list for `aov_ez`. I fixed this manually in `aov_car.Rd`, but I suspect that will not "stick". I think you may need to add some `#!` comments before that function to get its attention.

singmann commented:
Thanks, added!!
